cachi2/core/package_managers/yarn/main.py

```diff
@@ -40,14 +40,16 @@ def _resolve_yarn_project(project: Project, output_dir: RootedPath) -> list[Comp
 
     _configure_yarn_version(project)
 
+    if project.is_zero_installs:
+        raise PackageRejected(
+            ("Yarn zero installs are unsupported by cachi2"),
+            solution="Please convert your project to regular install-based workflow",
```

Comment:
I'm all for ADRs, though I'm not convinced we want to go as far as linking the corresponding documents in code. You know, URIs can change, projects move in between git forge services...I think the important part is to **have** an ADR :).

Reply:
We do have the [mechanism](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/errors.py#L18) in place already to link to docs, though. The reasoning here is that the error messages should lead the user to the solution, since we still struggle a lot to keep up with user questions. Cachito was not a user-friendly service, and we felt we needed to do better with Cachi2.

Maybe linking the ADR itself is not needed, but I definitely feel like we should link up relevant upstream docs once we have them, even at the risk of breaking some at some point.

Reply:
> We do have the [mechanism](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/errors.py#L18) in place already to link to docs, though. The reasoning here is that the error messages should lead the user to the solution, since we still struggle a lot to keep up with user questions. Cachito was not a user-friendly service, and we felt we needed to do better with Cachi2.

While I definitely agree this is both a fresh and noble approach towards end users wrt/ helpful error messages, however, the way it currently works is flawed for 2 reasons:

1. Links rot (that's a sad fact). It's so incredibly easy to miss the fact a link should be updated in a PR if the corresponding part of the code hasn't been touched, IOW there's only transitive connection to the docs between a link contained in an error message and the code being changed. That way, one has to be very diligent during review and dig through the code to make sure the docs part of error messages are always relevant and being accordingly updated.
2. (and this is a much bigger issue than the previous one) currently all the links point to the latest main branch contents IOW there's no versioning, so once we make a significant change either to the docs or we change some crucial functionality, older versions of cachi2 deployments' error messages would still point to the upstream docs which may have already been changed significantly due to a functional change coming in e.g. the next cachi2 major  version, so the docs attribute has just become completely irrelevant and confusing for such users (we must not assume everyone is always using upstream, even the container images are not re-pulled without an extra flag locally).

That said, I can add something to remain consistent with the rest of the code for now, but we'd still have to reconsider our approach given the problems mentioned ^above; depends on how strong your preference is @brunoapimentel @taylormadore

Reply:
I think we had accounted and decided to live with 1, but we probably overlooked 2. In any case, we don't currently have yarn docs, so you don't need to change anything ¯\_ (ツ)_/¯

Reply:
After doing a few tests locally, I feel like we can write a bigger solution message, that roughly tells the user to:
- if he's using pnp:
  - remove any commited cache files from the repo
  - run `yarn cache clean` in case he's running it locally and does not have commited cache
- otherwise:
  - remove the node_modules folder
  
Solution messages are expected to be bigger than the reason, or at least that's what we've been doing so far.